### PR TITLE
Fix AsyncioModbusSerialClient (serial options and create_serial_connection usage)

### PR DESCRIPTION
**Pull request by cbergmiller**
The API of serial_asyncio.create_serial_connection is different from the async socket API.
Also, the serial kwargs (baudrate etc.) were missing in AsyncioModbusSerialClient.__init__ Method.

### CODE REVIEW

File: pymodbus/client/async/asyncio/__init__.py [hunk 1]
```diff
@@ -707,11 +700,18 @@ def connect(self):
         """
         _logger.debug('Connecting.')
         try:
-            yield from self.loop.create_connection(self._create_protocol)
-            _logger.info('Connected to %s:%s.' % (self.host, self.port))
+            from serial_asyncio import create_serial_connection
```
Comment (dhoomakethu) [inline pymodbus/client/async/asyncio/__init__.py]: Please move this change to serial factory, https://github.com/cbergmiller/pymodbus/blob/fix/serial_async_client/pymodbus/client/async/factory/serial.py#L82.

Comment (cbergmiller) [reply pymodbus/client/async/asyncio/__init__.py]: Thanks for the review. I updated the pull request.

**Comment by dhoomakethu** [comment]
@cbergmiller the tests are failing on python3 , please have a look.

File: pymodbus/client/async/asyncio/__init__.py [hunk 2]
```diff
@@ -680,18 +680,27 @@ def __init__(self, port, protocol_class=None, framer=None,  loop=None,
...
-        self._connected = False
+        self._connected_event = asyncio.Event()
 
     def stop(self):
         """
         Stops connection
         :return:
         """
-        if self._connected:
+        if self._connected.is_set():
```
Comment (dhoomakethu) [inline pymodbus/client/async/asyncio/__init__.py]: You can directly use the property `self._connected` here.

File: pymodbus/client/async/asyncio/__init__.py [hunk 3]
```diff
@@ -718,8 +725,8 @@ def protocol_made_connection(self, protocol):
         Protocol notification of successful connection.
         """
         _logger.info('Protocol made connection.')
-        if not self._connected:
-            self._connected = True
+        if not self._connected_event.is_set():
```
Comment (dhoomakethu) [inline pymodbus/client/async/asyncio/__init__.py]: You can directly use the property `_connected` here.

Comment (cbergmiller) [reply pymodbus/client/async/asyncio/__init__.py]: I didn't want to break compatibility with possible subclasses or mixins. So i created the `_connected` property that returns the state of the connected event. But you are right it is cleaner to use the `_property` inside the class as well. I will update the pull request.

**Pull request merged**